[tyx021] iP

tyx021 opened the pull request:
### Duke Chatting Bot

> ~~“Cats rule the world.” – Jim Davis~~

Why Duke? Because it is
- text-based
- easy to use
- **FREE**

All you need to to is:

1. download _ip.jar_ from [here](https://github.com/tyx021/ip/releases/tag/v0.1).
2. double-click to run it.
3. add your tasks.
4. let it manage your tasks for you :smile_cat:

Features to be implemented:
- [x] Text-based UI
- [ ] GUI

You can practice your Java with it. Here's the `main` method:
```java
public static void main(String[] args) {
    new Duke("data/duke.txt", "/data").run();
}
```

Commit:
Add Gradle support

tyx021 committed:
Added Greet, Echo, Exit

tyx021 committed:
Add text to List

tyx021 committed:
Added Task class

tyx021 committed:
Added mark & unmark tasks

tyx021 committed:
Created Todo, Deadline, Event class

tyx021 committed:
Add support for tracking tasks of type ToDos, Events and Deadlines

tyx021 committed:
Added Text UI Testing

tyx021 committed:
Handle errors with Exception

tyx021 committed:
Rename exception as DukeException

tyx021 committed:
Added support for deleting tasks

tyx021 committed:
Added use of Enums
Handled IllegalArgumentException caused by Enums

tyx021 committed:
Fix indentation

tyx021 committed:
Fixs some coding style

tyx021 committed:
Adds save and load tasks

tyx021 committed:
Adds date and time to Deadline and Event object

tyx021 committed:
Merge branch 'branch-Level-7'

tyx021 committed:
Merge branch 'branch-Level-8'

tyx021 committed:
Uses More OOP
Extract the following classes:
- Ui
- Storage
- Parser
- TaskList

tyx021 committed:
Refactor packages

tyx021 committed:
Merge branch 'add-gradle-support' of https://github.com/tyx021/ip

tyx021 committed:
Adds some dummy test

tyx021 committed:
Packages Duke as a JAR file

tyx021 committed:
Adds gradle file to gitignore

tyx021 committed:
Ignores gradle file

tyx021 committed:
Adds JavaDoc

tyx021 committed:
Follows the coding standard

tyx021 committed:
Adds seraching task by keyword

tyx021 committed:
Merge branch 'branch-A-JavaDoc'

tyx021 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java

tyx021 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/exception/DukeException.java

seadragon2000341 commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,65 @@
+ */
+public class Deadline extends Task {
+
+    private final LocalDate by;
```
Suggest to change the name to more be more meaningful instead of "by"

seadragon2000341 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     * @param command Users' inputs
+     */
+    public void parse(String command) {
+        String[] s;
```
Possibly change to a more meaningful name as well?

seadragon2000341 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+        }
+    }
+    
+    enum Actions {LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND}
```
Great categorization

seadragon2000341 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,82 @@
+    /**
+     * Indicates the completion status of the task.
+     */
+    boolean isDone;
```
Nice that you have followed convention

seadragon2000341 reviewed (commented):
overall, great work in following the coding standards!

tyx021 committed:
Add CheckStyle with gradle

tyx021 committed:
Fix coding style violations spotted by checkStyle

tyx021 committed:
Merge branch 'branch-A-CheckStyle'

tyx021 commented:
Thanks @seadragon2000341 for your review! I will try to replace them with better variable names! :)

tyx021 committed:
Add GUI for Duke

tyx021 committed:
Merge branch 'branch-Level-10'

tyx021 committed:
Adds background color to layout

tyx021 committed:
Fix style.

tyx021 committed:
Clean unused methods.

tyx021 committed:
Fix bug for Find command

Find command will modify data of tasks.

When we perform a find command, current task list will be replaced by
filtered task list because Java built-in method filter() was used
originally.

Changes:
* Instead of returning the filtered task list, the method will return a
string representation of filtered task list.
* New implementation will not modify original data.

tyx021 committed:
Fix javaDoc and coding style.

tyx021 committed:
Use Varargs in TaskList::findMatchingTasks

tyx021 committed:
Add `assert` to `TaskList.java` to document some assumptions.

tyx021 committed:
Make use of SLAP to reduce method length for `Parser::parse`
Extract out the part of handling different commands as different methods.

tyx021 committed:
Use Streams feature to help printing out list of tasks

tyx021 committed:
Merge pull request #2 from tyx021/A-Assertions

Add `assert` to `TaskList.java` to document some assumptions.

tyx021 committed:
Merge pull request #3 from tyx021/A-CodeQuality

Make use of SLAP to reduce method length for `Parser::parse`

tyx021 committed:
Merge branch 'master' into A-Streams

tyx021 committed:
Merge pull request #4 from tyx021/A-Streams

Use Streams feature to help printing out list of tasks

tyx021 committed:
Add gradle.yml

tyx021 committed:
Adds Sort feature:
Tasks will be sorted based on their completion status, then by the
lexicographic order of task description

tyx021 committed:
Merge branch 'C-Sort'

tyx021 committed:
Improve GUI.
- Duke and user have different color of dialog box
- Prevent long message being shown as ellipses

tyx021 committed:
Improve on Duke's response.
- Greets user at the beginning
- Closes the application when user inputs "bye"

tyx021 committed:
Adds JavaDoc and fix coding standards.

tyx021 committed:
Improve GUI

tyx021 committed:
Update README.md

tyx021 committed:
Fix javafx version

tyx021 committed:
Fix version issues

tyx021 committed:
Fix CI workflow
